src/db.rs

```diff
@@ -108,12 +137,19 @@ impl Storage for DB {
                 String::from_utf8_lossy(key),
             ));
         }
-        go_result
+        go_result.and(Ok(used_gas))
```

Comment:
fair enough, I think you convinced me we don't care about gas on FfiError, as any charge there is known to the sdk/runtime anyway and we will abort so no need to track it in rust

Reply:
:smile: